### PR TITLE
library/dao/playlistdao: Increase robustness by using prepared statements

### PR DESCRIPTION
**Pull request by Holzhaus**

### CODE REVIEW

File: src/library/dao/playlistdao.cpp [hunk 1]
```diff
@@ -397,24 +397,18 @@ void PlaylistDAO::removeHiddenTracks(const int playlistId) {
...
-    }
-
+    query.prepare(QStringLiteral(
+            "SELECT p1.position FROM PlaylistTracks p1"
+            "WHERE p1.id NOT IN ("
+            "SELECT p2.id FROM PlaylistTracks p2"
+            "INNER JOIN library ON library.id = p2.track_id "
+            "WHERE p2.playlist_id = p1.playlist_id "
+            "AND library.mixxx_deleted = 0 ) "
+            "AND p1.playlist_id=:id"));
```
Comment (Holzhaus) [inline src/library/dao/playlistdao.cpp:408]: No super sure about this one, please verify if this is correct.

File: src/library/dao/playlistdao.cpp [hunk 2]
```diff
@@ -819,36 +815,47 @@ void PlaylistDAO::moveTrack(const int playlistId, const int oldPosition, const i
...
+    }
+    if (!query.exec()) {
+        LOG_FAILED_QUERY(query);
     }
-    query.exec(queryString);
 
-    query.exec(QStringLiteral(
-            "UPDATE PlaylistTracks SET position = %1 "
+    query.prepare(QStringLiteral(
+            "UPDATE PlaylistTracks SET position=:mew_position "
```
Comment (uklotzde) [inline src/library/dao/playlistdao.cpp:851]: typo

**Comment by uklotzde** [comment]
Fixes https://bugs.launchpad.net/mixxx/+bug/1884356

Thanks for jumping in, Jan!

File: src/library/dao/playlistdao.cpp (hunk 1 above)
Comment (uklotzde) [reply src/library/dao/playlistdao.cpp:408]: I'll check and test this later.

File: src/library/dao/playlistdao.cpp (hunk 2 above)
Comment (Holzhaus) [reply src/library/dao/playlistdao.cpp]: oops, sorry

File: src/library/dao/playlistdao.cpp [hunk 3]
```diff
@@ -397,24 +397,18 @@ void PlaylistDAO::removeHiddenTracks(const int playlistId) {
...
-                    "AND PlaylistTracks.playlist_id = %1")
-                    .arg(playlistId);
-
     QSqlQuery query(m_database);
-    if (!query.prepare(queryString)) {
-        LOG_FAILED_QUERY(query);
-        return;
-    }
-
+    query.prepare(QStringLiteral(
```
Comment (uklotzde) [inline src/library/dao/playlistdao.cpp:401]: MUCH (25 ms vs. 1000 ms) faster:

```sql
SELECT PlaylistTracks.position FROM PlaylistTracks
INNER JOIN library ON library.id=PlaylistTracks.track_id
AND library.mixxx_deleted=0
AND PlaylistTracks.playlist_id=:id
```

Comment (uklotzde) [reply src/library/dao/playlistdao.cpp:401]: :+1:  Works as is. No need to change it.

**Comment by uklotzde** [comment]
LGTM

**Pull request merged**